src/tn/eduVision/services/MaterielService.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package tn.eduVision.services;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import tn.eduVision.entités.Materiel;
+import tn.eduVision.entités.TypeRessource;
+import tn.eduVision.exceptions.NoDataFoundException;
+import tn.eduVision.tools.CustomLogger;
+import tn.eduVision.tools.SqlConnectionManager;
+
+/**
+ *
+ * @author job_j
+ */
+public class MaterielService implements Iservices<Materiel>{
+
+     private final Connection _connection = SqlConnectionManager.getConnection();
+     private static final Logger _logger = CustomLogger.getInstance().getLogger();
+     private static PreparedStatement statement = null;
+    
+    @Override
```

Comment:
why override ?

Reply:
@SeifEddineJridi when i auto impliment from the interface tji b @override so mich ena zedtha won't be a problem